roomserver/internal/input/input_latest_events.go

```diff
@@ -296,11 +298,13 @@ func (u *latestEventsUpdater) calculateLatest(
 	referenced, err := u.updater.IsReferenced(newEvent.EventReference)
 	if err != nil {
 		logrus.WithError(err).Errorf("Failed to retrieve event reference for %q", newEvent.EventReference.EventID)
+		return fmt.Errorf("u.updater.IsReferenced (new): %w", err)
```

Comment:
Why are we now returning from here but didn't before?

Reply:
I put this in because it feels like a genuine error condition, e.g. we failed to talk to the database or something. It doesn't feel right to just fall through and possibly end up with a wrong state snapshot.